app/controllers/api/v2/stars_controller.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+module Api
+  module V2
+    class StarsController < RestfulController
+      skip_before_action :load_resource
+
+      def create
+        @map = Map.find(params[:id])
+        @star = Star.new(user: current_user, map: @map)
+        authorize @map, :star?
+        create_action
+
+        if @star.errors.empty?
+          render json: @map, scope: default_scope, serializer: MapSerializer, root: serializer_root
```

Comment:
In a theoretical world, this should actually return an instance of a star resource... but that's kinda dumb, so I decided to return the new map instead